src/main/java/filemanager/directorytracker/ScheduleFileDirectory.java

```diff
@@ -91,30 +61,39 @@ public void goThroughToCheckFile(Command command) {
                         .filter((file) -> Files.isRegularFile(file) && matchPattern(file.toString()))
                         .forEach(path -> {
                             try {
-                                readService.readJson(new FileInputStream(path.toString()), interactions);
+                                interactions.addAll(readService.readJson(new FileInputStream(path.toString())));
                             } catch (IOException e) {
                                 e.printStackTrace();
                             }
                         });
             } catch (IOException e) {
                 e.printStackTrace();
             }
-
             groupService.groupByClient(interactions, interactionsMap);
```

Comment:
Please follow same approach are you did for ```readService.readJson``` so that this function had no side-effects.